### ISSUE TITLE
AsyncLocal no longer works in V3

### ISSUE THREAD

**Issue by SimonCropp** [xunit/xunit#3033]
This test passes in v2 but fails in v3
```
public class Tests
{
    static AsyncLocal<object> asyncLocal = new();

    public Tests() =>
        asyncLocal.Value = new();

    [Fact]
    public void Test() =>
        Assert.NotNull(asyncLocal.Value);
}
```

repro https://github.com/SimonCropp/xunitV3AsyncLocal

**Comment by bradwilson** [comment]
I should've anticipated this when fixing #3014.

![giphy](https://github.com/user-attachments/assets/02f72384-cb9a-4ed4-9e85-c677e3f77f4e)

**Comment by bradwilson** [comment]
Fixed in `0.5.0-pre.4` https://xunit.net/docs/using-ci-builds

I bumped the version because it breaks an API contract in `TestRunner`.

**Comment by SimonCropp** [comment]
thanks
